### DISCUSSION TITLE
"The user denied permission to use Service Worker." when using codespaces

### DISCUSSION THREAD

**Discussion by vadi2** [community/community#26316] [Codespaces]
<p>I’m still experiencing the error reported in <a href="https://github.community/t/error-loading-webview/134455">September 2020</a> with the permission error -</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/d/0/d0d0c4ac88fecbdf7db45ade9ae520eada8bc226.png" data-download-href="/uploads/short-url/tNgB3SkVm6we2WXTXV8e3yixRMq.png?dl=1" title="Screenshot 2021-08-13 7.49.05 AM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/110988/181096870-a402e107-05ae-4c22-bb20-2a65386f239e.png" alt="Screenshot 2021-08-13 7.49.05 AM" data-base62-sha1="tNgB3SkVm6we2WXTXV8e3yixRMq" width="600" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/d/0/d0d0c4ac88fecbdf7db45ade9ae520eada8bc226_2_600x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/d/0/d0d0c4ac88fecbdf7db45ade9ae520eada8bc226_2_900x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/d/0/d0d0c4ac88fecbdf7db45ade9ae520eada8bc226_2_1200x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/d/0/d0d0c4ac88fecbdf7db45ade9ae520eada8bc226_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2021-08-13 7.49.05 AM</span><span class="informations">1518×1265 165 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>The permissions do seem to be blocked, but I didn’t do it on purpose. Furthermore, even allowing them once doesn’t really work because the URL of the service worker changes every time.</p>
<p>How to fix?</p>

**Comment by nebbles** [comment]
<p>I am also getting this error. For me, it’s when trying to use <em>Markdown: Open Preview</em> command. I tried allowing the cookie manually through the GUI on Brave, but after refreshing the page I kept seeing the same issue.</p>
<p>Edit: I have just run into the same error again when trying to display the <em>Create Pull Request</em> panel for the <em>GitHub Pull Request</em> extension.</p>

**Comment by rob4226** [comment]
<p>I am getting the same error using Brave. I tried turning off all the shields but still got the error.</p>
<pre><code class="lang-auto">Error loading webview: Error: Could not register service workers: NotSupportedError: Failed to register a ServiceWorker for scope
('https://21b296a2-9664-475d-8387-f0fb4fef5697.vscode-webview.net/stable/3866c3553be8b268c8a7f8c0482c0c0177aa8bfa/out/vs/workbench/contrib/webview/browser/pre/') with script ('https://21b296a2-9664-475d-8387-f0fb4fef5697.vscode-webview.net/stable/3866c3553be8b268c8a7f8c0482c0c0177aa8bfa/out/vs/workbench/contrib/webview/browser/pre/service-worker.js?id=21b296a2-9664-475d-8387-f0fb4fef5697&amp;swVersion=2&amp;extensionId=vscode.markdown-language-features&amp;platform=browser&amp;vscode-resource-base-authority=vscode-resource.vscode-webview.net'):
The user denied permission to use Service Worker..
</code></pre>

**Reply by danielaltamirano1993** [reply]
December 2022 : Is true. And this is the solution. Confirmed!  Turn Off the Brave Shields in Codespaces

**Comment by JarLob** [comment, marked as answer]
<p>Most probably you as me have “Block third-party cookies” setting enabled. I was able to workaround the error by adding <code>[*.]github.dev</code> to “Sites that can always use cookies” with a checkbox “Including third-party cookies on this site” selected.</p>

**Comment by ahmed002255** [comment]
<p>oki  you can solve this problem by making sure that you browser dont block the cookies is the web site you can do it by open sitting then  open site sitting then make sure that you allow the browser to use cookies and enjoy cooding</p>

**Comment by serbanradulescu** [comment]
<p>Lots of thanks !!! I went on the forum and it drove me to a rabbit hole, just duplicates of the issue over duplicates</p>

**Reply by ghost** [reply]
I'm in chrome. Couldn't fix it even after allowing to use cookies.

**Comment by mbjoe0** [comment]
I'm facing this error:

root@codespaces-b20630:/workspaces/meltano-codespace-ready# ./meltano_tut select_db
=== Running select DB ===
Traceback (most recent call last):
  File "/workspaces/meltano-codespace-ready/codespaces_tutorial/select_duckdb.py", line 4, in <module>
    con = duckdb.connect(database='output/my.duckdb', read_only=True)
duckdb.IOException: IO Error: Trying to read a database file with version number 64, but we can only read version 51.
The database file was created with an newer version of DuckDB.

The storage of DuckDB is not yet stable; newer versions of DuckDB cannot read old database files and vice versa.
The storage will be stabilized when version 1.0 releases.

For now, we recommend that you load the database file in a supported version of DuckDB, and use the EXPORT DATABASE command followed by IMPORT DATABASE on the current version of DuckDB.

See the storage page for more information: https://duckdb.org/internals/storage

**Comment by deveshv19** [comment]
In m case AniVirus's (Anti-Banner) was causing issue. I had to turn that feature off in order for this to work ( :( ). No other option worked. I think there are some unsafe css/js scripts and the Anti-Virus thinks that it is a threat. The best way to debug/cross-verify this is by looking at browser console (In case you are opening CodeSpaces via browser).
